MahApps.Metro/Controls/RangeSlider.cs

```diff
@@ -2020,22 +2020,22 @@ private void SnapToTickHandle(ButtonType type, Direction mDirection, Double diff
         }
 
         //Calculating next value for Tick
-        private Double CalculateNextTick(Direction dir, double chekingValue, double distance, bool moveDirectlyToNextTick)
+        private Double CalculateNextTick(Direction dir, double checkingValue, double distance, bool moveDirectlyToNextTick)
         {
+            double checkingValuePos = checkingValue - Minimum;
             if (!IsMoveToPointEnabled)
             {
                 //Check if current value is exactly Tick value or it situated between Ticks
-                if (!IsDoubleCloseToInt((chekingValue - Minimum) / TickFrequency))
+                double checkingValueChanged = checkingValuePos + distance;
+                double x = checkingValueChanged / TickFrequency;
+                if (!IsDoubleCloseToInt(x))
                 {
-                    double x = (chekingValue - Minimum) / TickFrequency;
                     distance = TickFrequency * (int)x;
                     if (dir == Direction.Increase)
                     {
                         distance += TickFrequency;
                     }
-                    distance = (distance - Math.Abs(chekingValue - Minimum));
-                    _currenValue = 0;
-                    return Math.Abs(distance);
```

Comment:
I am unsure. Isn't the code around line 2053 and onwards valid for this case as well?